plugins/kubernetes/src/error-detection/common.ts

```diff
@@ -14,59 +14,15 @@
  * limitations under the License.
  */
 
-import {
-  DetectedError,
-  ErrorDetectable,
-  ErrorDetectableKind,
-  ErrorMapper,
-} from './types';
+import { DetectedError, ErrorMapper } from './types';
 
 // Run through the each error mapper for each object
 // returning a deduplicated (mostly) result
-export const detectErrorsInObjects = <T extends ErrorDetectable>(
+export const detectErrorsInObjects = <T>(
   objects: T[],
-  kind: ErrorDetectableKind,
-  clusterName: string,
   errorMappers: ErrorMapper<T>[],
 ): DetectedError[] => {
-  // Build up a map of errors
-  // key: the joined message produced by an error
-  // value: the error
-  const errors = new Map<string, DetectedError>();
-
-  for (const object of objects) {
-    for (const errorMapper of errorMappers) {
-      if (errorMapper.errorExists(object)) {
-        const message = errorMapper.messageAccessor(object);
-
-        // TODO This is not perfect as errors with uuid/hashes/date/times will not be caught by this
-        const dedupKey = message.join('');
-
-        const value = errors.get(dedupKey);
-
-        const name = object.metadata?.name ?? 'unknown';
-        const namespace = object.metadata?.namespace ?? 'unknown';
-
-        if (value !== undefined) {
-          // This gets translated into the Chip "+5 others"
-          // in the ErrorReporting component
-          // but we need to keep the names so we can easily
-          // find which objects owns the error later
-          value.names.push(name);
-          errors.set(dedupKey, value);
-        } else {
-          errors.set(dedupKey, {
-            cluster: clusterName,
-            kind: kind,
-            names: [name],
-            message: message,
-            severity: errorMapper.severity,
-            namespace,
-          });
-        }
-      }
-    }
-  }
-
-  return Array.from(errors.values());
+  return errorMappers.flatMap(errorMapper => {
+    return objects.flatMap(o => errorMapper.detectErrors(o));
+  });
```

Comment:
Feels like objects -> mappers would be the more intuitive order? Although perhaps the ordering here doesn't matter

Reply:
Really I thought the opposite was more intuitive 😂  I can change this

Reply:
done